atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/ConflictDetectionManagers.java

```diff
@@ -54,34 +52,23 @@ public static ConflictDetectionManager withoutConflictDetection(KeyValueService
         return new ConflictDetectionManager(getNoConflictDetectSupplier(kvs));
     }
 
-    private static RecomputingSupplier<Map<TableReference, ConflictHandler>> getNoConflictDetectSupplier(final KeyValueService kvs) {
-        return RecomputingSupplier.create(new Supplier<Map<TableReference, ConflictHandler>>() {
-            @Override
-            public Map<TableReference, ConflictHandler> get() {
-                Set<TableReference> tables = kvs.getAllTableNames();
-                return Maps.asMap(tables, Functions.constant(ConflictHandler.IGNORE_ALL));
-            }
-        });
+    private static Supplier<Map<TableReference, ConflictHandler>> getNoConflictDetectSupplier(final KeyValueService kvs) {
+        return () -> {
+            Set<TableReference> tables = kvs.getAllTableNames();
+            return Maps.asMap(tables, Functions.constant(ConflictHandler.IGNORE_ALL));
+        };
     }
 
-    private static RecomputingSupplier<Map<TableReference, ConflictHandler>> getTablesToConflictDetectSupplier(final KeyValueService keyValueService) {
-        return RecomputingSupplier.create(new Supplier<Map<TableReference, ConflictHandler>>() {
-            @Override
-            public Map<TableReference, ConflictHandler> get() {
-                return getTablesToConflictDetect(keyValueService);
-            }
-        });
+    private static Supplier<Map<TableReference, ConflictHandler>> getTablesToConflictDetectSupplier(final KeyValueService keyValueService) {
+        return () -> getTablesToConflictDetect(keyValueService);
     }
 
     private static Map<TableReference, ConflictHandler> getTablesToConflictDetect(KeyValueService kvs) {
-        return ImmutableMap.copyOf(Maps.transformValues(kvs.getMetadataForTables(), new Function<byte[], ConflictHandler>() {
-            @Override
-            public ConflictHandler apply(byte[] metadataForTable) {
-                if (metadataForTable != null && metadataForTable.length > 0) {
-                    return TableMetadata.BYTES_HYDRATOR.hydrateFromBytes(metadataForTable).getConflictHandler();
-                } else {
-                    return ConflictHandler.RETRY_ON_WRITE_WRITE;
-                }
+        return ImmutableMap.copyOf(Maps.transformValues(kvs.getMetadataForTables(), metadataForTable -> {
+            if (metadataForTable != null && metadataForTable.length > 0) {
+                return TableMetadata.BYTES_HYDRATOR.hydrateFromBytes(metadataForTable).getConflictHandler();
+            } else {
+                return ConflictHandler.RETRY_ON_WRITE_WRITE;
```

Comment:
This is not your fault, but we should have a log.warn here that says "hey, I couldn't find any metadata for your table {tablename} so I'm defaulting to RETRY_ON_WRITE_WRITE for it; this may break your application or ruin guarantees you thought you had"